cmd/whitesourceExecuteScan.go

```diff
@@ -76,13 +76,62 @@ func (w *whitesourceUtilsBundle) FileOpen(name string, flag int, perm os.FileMod
 }
 
 func (w *whitesourceUtilsBundle) GetArtifactCoordinates(buildTool, buildDescriptorFile string, options *versioning.Options) (versioning.Coordinates, error) {
+	if err := validationBuildDescriptorFile(buildTool, buildDescriptorFile); err != nil {
+		return versioning.Coordinates{}, err
+	}
 	artifact, err := versioning.GetArtifact(buildTool, buildDescriptorFile, options, w)
 	if err != nil {
 		return versioning.Coordinates{}, err
 	}
 	return artifact.GetCoordinates()
 }
 
+func validationBuildDescriptorFile(buildTool, buildDescriptorFile string) error {
+	if buildDescriptorFile == "" {
+		return nil
+	}
+	switch buildTool {
+	case "dub":
+		if filepath.Ext(buildDescriptorFile) != ".json" {
+			return errors.New("extension of buildDescriptorFile must be in '*.json'")
+		}
+	case "gradle":
+		if filepath.Ext(buildDescriptorFile) != ".properties" {
+			return errors.New("extension of buildDescriptorFile must be in '*.properties'")
+		}
+	case "golang":
+		if !strings.HasSuffix(buildDescriptorFile, "go.mod") &&
```

Comment:
probably for the files with extensions, it's better to use filepath.Ext()

Reply:
In this particular case, using filepath.Ext() is not suitable because we need to specifically check for "go.mod", instead of just the ".mod" extension. This is a requirement for Go projects which use modules, ensuring that the file name exactly matches "go.mod".